packages/block-library/src/block/edit.native.js

```diff
@@ -0,0 +1,262 @@
+import { help } from '@wordpress/icons';
+import { requestUnsupportedBlockFallback } from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import EditTitle from './edit-title';
+
+class ReusableBlockEdit extends Component {
+	constructor( { reusableBlock } ) {
+		super( ...arguments );
+
+		this.setBlocks = this.setBlocks.bind( this );
+		this.toggleSheet = this.toggleSheet.bind( this );
+		this.requestFallback = this.requestFallback.bind( this );
+
+		if ( reusableBlock ) {
+			// Start in edit mode when we're working with a newly created reusable block
+			this.state = {
+				// Since edition is not supported yet isEditing is always false
+				isEditing: false,
+				blocks: parse( reusableBlock.content ),
+				showHelp: false,
+				sendFallbackMessage: false,
+			};
+		} else {
+			// Start in preview mode when we're working with an existing reusable block
+			this.state = {
+				isEditing: false,
+				blocks: [],
+				showHelp: false,
+				sendFallbackMessage: false,
+			};
+		}
+	}
+
+	componentDidMount() {
+		this.props.fetchReusableBlock();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const hasBlockFetchCompleted =
+			! prevProps.reusableBlock && this.props.reusableBlock;
+		const hasBlockContentChanged =
+			prevProps.reusableBlock?.content !==
+			this.props.reusableBlock?.content;
+
+		if ( hasBlockFetchCompleted || hasBlockContentChanged ) {
+			this.setState( {
+				blocks: parse( this.props.reusableBlock.content ),
+			} );
+		}
+	}
+
+	toggleSheet() {
+		this.setState( {
+			showHelp: ! this.state.showHelp,
+		} );
+	}
+
+	requestFallback() {
+		this.toggleSheet();
+		this.setState( { sendFallbackMessage: true } );
+	}
+
+	setBlocks( blocks ) {
+		this.setState( { blocks } );
+	}
+
+	renderSheet() {
+		const { reusableBlock } = this.props;
+		const { showHelp } = this.state;
+
+		const { getStylesFromColorScheme, clientId } = this.props;
+		const infoTextStyle = getStylesFromColorScheme(
+			styles.infoText,
+			styles.infoTextDark
+		);
+		const infoTitleStyle = getStylesFromColorScheme(
+			styles.infoTitle,
+			styles.infoTitleDark
+		);
+		const infoSheetIconStyle = getStylesFromColorScheme(
+			styles.infoSheetIcon,
+			styles.infoSheetIconDark
+		);
+		const actionButtonStyle = getStylesFromColorScheme(
+			styles.actionButton,
+			styles.actionButtonDark
+		);
+
+		const infoTitle =
+			Platform.OS === 'android'
+				? __(
+						"Reusable blocks aren't editable on WordPress for Android"
+				  )
+				: __( "Reusable blocks aren't editable on WordPress for iOS" );
+
+		return (
+			<BottomSheet
+				isVisible={ showHelp }
+				hideHeader
+				onClose={ this.toggleSheet }
+				onModalHide={ () => {
+					if ( this.state.sendFallbackMessage ) {
+						// On iOS, onModalHide is called when the controller is still part of the hierarchy.
+						// A small delay will ensure that the controller has already been removed.
+						this.timeout = setTimeout( () => {
```

Comment:
I see this is a pattern we used elsewhere, so it's likely somewhat safe. Not essential, but if you have any thoughts on how this can be improved to rely on a something state update instead of the delay, would be great. But again, not essential, looks like we felt this was good for now originally https://github.com/WordPress/gutenberg/pull/21150/files#r429455647

Reply:
I agree this is not the best, but was needed to present the modal view on iOS safely.
A proper solution I found needed modification of the Modal component.
I'm happy to have a different solution, or to wrap this code on a different component so can be easily shared.

No need to do this on this PR though. Just wild ideas :)

Reply:
Yeah I had to deal with this issue on other React native projects, unfortunately I haven't seen yet an elegant solution apart from isolating it in a component. In the project I'm currently working we implemented our own custom modal because we had a lot of issues with them like this one, we use `react-navigation` and since you can control the transitions we rendered them as screens and we imitate the behavior of a native modal 😄 .

Regarding possible options, in my opinion the timeout option is not too bad. Using the state could be another one but it would imply to add an extra render, usually I try to avoid unnecessary renders if they're not gonna really modify the UI but it depends on the case.

Reply:
We discussed this (offline) when it was used for the UBE project, a little more info [here](https://github.com/WordPress/gutenberg/pull/21150#discussion_r429461377) FWIW.